### PR TITLE
Local Persistence

### LINKED ISSUE #98: CoreData for local storage
- [x] Documents 
- [x] Annotations
- [x] AnnotationText 
- [x] AnnotationHandwriting 
- [x] SelectionBox 
- [ ] DocumentShare (No need to store this locally)

### PR DESCRIPTION
**Pull request by sivayogasubramanian**
Closes https://github.com/Annotato/annotato/issues/98

### Note

1. When data arrives at the core data layer, it would always be valid with all the timestamps set appropriately.
2. We do not delete anything from core data. Instead, we just set `deletedAt` to some date value.

#### Document

- [x] getOwnDocuments returns own documents
- [x] getOwnDocuments does not return deleted documents
- [x] getSharedDocuments returns shared documents
- [x] getSharedDocuments does not return deleted documents
- [x] getDocument returns document
- [x] getDocument should not return deleted document
- [x] createDocument creates document
- [x] updateDocument/deleteDocument updates document (we are just updating deletedAt for delete)

#### Annotation

- [x] createAnnotation creates annotation
- [x] updateAnnotation/deleteAnnotation updates annotation (we are just updating deletedAt for delete)

### CODE REVIEW

File: Annotato/Annotato/Persistence/LocalPersistenceManager/AnnotatoLocal.xcdatamodeld/AnnotatoLocal.xcdatamodel/contents [hunk 1]
```diff
@@ -43,21 +44,23 @@
...
+        <attribute name="endPointX" attributeType="Double" defaultValueString="0.0" usesScalarValueType="YES"/>
+        <attribute name="endPointY" attributeType="Double" defaultValueString="0.0" usesScalarValueType="YES"/>
         <attribute name="id" attributeType="UUID" usesScalarValueType="NO"/>
-        <attribute name="recipientId" attributeType="String"/>
+        <attribute name="startPointX" attributeType="Double" defaultValueString="0.0" usesScalarValueType="YES"/>
+        <attribute name="startPointY" attributeType="Double" defaultValueString="0.0" usesScalarValueType="YES"/>
         <attribute name="updatedAt" optional="YES" attributeType="Date" usesScalarValueType="NO"/>
-        <relationship name="documentEntity" optional="YES" maxCount="1" deletionRule="Nullify" destinationEntity="DocumentEntity" inverseName="documentShareEntities" inverseEntity="DocumentEntity"/>
+        <relationship name="annotationEntity" optional="YES" maxCount="1" deletionRule="Nullify" destinationEntity="AnnotationEntity" inverseName="selectionBoxEntity" inverseEntity="AnnotationEntity"/>
```
Comment (ganhongyao) [inline Annotato/Annotato/Persistence/LocalPersistenceManager/AnnotatoLocal.xcdatamodeld/AnnotatoLocal.xcdatamodel/contents:57]: I think we may need to remove this in future if we choose to have selection boxes that can belong to different entities? But okay for now

**Review by ganhongyao — COMMENTED** [review]
Will test more thoroughly once #115 is up

File: Annotato/Annotato/Persistence/LocalPersistenceManager/AnnotatoLocal.xcdatamodeld/AnnotatoLocal.xcdatamodel/contents (hunk 1 above)
Comment (sivayogasubramanian) [reply Annotato/Annotato/Persistence/LocalPersistenceManager/AnnotatoLocal.xcdatamodeld/AnnotatoLocal.xcdatamodel/contents:57]: In core data I think that would just be another relationship.

**Review by ganhongyao — APPROVED** [review]
LGTM, can merge once the TODOs are cleaned up. Will do integration in another PR

**Pull request merged**